pkg/scheduler/scheduling_application.go

```diff
@@ -408,6 +408,19 @@ func (sa *SchedulingApplication) sortRequests(ascending bool) {
 	}
 }
 
+// update container scheduling state to shim if the plugin is registered
+func (sa *SchedulingApplication) updateContainerSchedulingStateIfNeeded(ask *schedulingAllocationAsk,
```

Comment:
The `IfNeeded` is not correct. There is no state or other check here we call always if the plugin is configured.
Should be `updateContainerSchedulingState`